include/bluetooth/services/cts_client.h

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#ifndef BT_CTS_CLIENT_H_
+#define BT_CTS_CLIENT_H_
+
+/** @file
+ *
+ * @defgroup bt_gatt_cts_c Current Time Service Client
+ * @{
+ *
+ * @brief Current Time Service Client module.
+ */
+
+#include <bluetooth/gatt.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/gatt_dm.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**@brief "Exact Time 256" field of the Current Time characteristic. */
+struct bt_cts_exact_time_256 {
+	uint16_t year;
+	uint8_t month;
+	uint8_t day;
+	uint8_t hours;
+	uint8_t minutes;
+	uint8_t seconds;
+	uint8_t day_of_week;
+	uint8_t fractions256;
+};
+
+/**@brief "Adjust Reason" field of the Current Time characteristic. */
+struct bt_cts_adjust_reason {
+	uint8_t manual_time_update : 1;
+	uint8_t external_reference_time_update : 1;
+	uint8_t change_of_time_zone : 1;
+	uint8_t change_of_daylight_savings_time : 1;
+};
+
+/**@brief Data structure for the Current Time characteristic. */
+struct bt_cts_current_time_char {
+	struct bt_cts_exact_time_256 exact_time_256;
+	struct bt_cts_adjust_reason adjust_reason;
+};
+
+/**@brief Current Time Service client event type. */
+enum bt_cts_client_evt_type {
+	/** A new Current Time reading has been received. */
+	BT_CTS_CLIENT_EVT_CURRENT_TIME,
+
+	/** The Current Time value received from the peer is invalid. */
+	BT_CTS_CLIENT_EVT_INVALID_TIME
+};
+
+/**@brief Current Time Service client event. */
+struct bt_cts_client_evt {
+	/** Type of event. */
+	enum bt_cts_client_evt_type evt_type;
+
+	/** Current Time characteristic data. This is filled when the evt_type is @ref BT_CTS_CLIENT_EVT_CURRENT_TIME. */
+	struct bt_cts_current_time_char current_time;
+};
+
+struct bt_cts_client;
+
+/**@brief Current Time Service client event handler type. */
+typedef void (*bt_cts_client_evt_handler)(struct bt_cts_client *cts_c,
+					  const struct bt_cts_client_evt *evt);
```

Comment:
I think I'd like to change the event handler to be just a read_cb supplied to the read_current_time function, instead of as an event handler for the module.
Also more often we have an error code in the callback, which could replace the evt_type in this case:
```
int bt_cts_client_init(struct bt_cts_client *cts);

typedef void (*bt_cts_read_cb)(struct bt_cts_client *cts, struct bt_cts_current_time_char *current_time, int err);`

int bt_cts_read_current_time(struct bt_cts_client *cts, bt_cts_read_cb_t func);
```

Reply:
Will do.